src/proxyp/Message.h

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (C) 1996-2018 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#ifndef SQUID_PROXYP_MESSAGE_H
+#define SQUID_PROXYP_MESSAGE_H
+
+#include "base/RefCount.h"
+#include "ip/Address.h"
+#include "proxyp/Elements.h"
+#include "sbuf/SBuf.h"
+
+namespace ProxyProtocol {
+
+/// PROXY protocol v1 or v2 message
+class Message: public RefCountable
+{
+public:
+    typedef RefCount<Message> Pointer;
+    typedef std::vector<Two::Tlv> Tlvs;
+
+    Message(const char *ver, const uint8_t cmd = Two::cmdProxy);
+
+    /// HTTP header-like string representation of the message.
+    /// The returned string has one line per pseudo header version,
```

Comment:
I see no reason to itemize pseudo headers here, especially since some of them may not be available. Let's simplify:

```suggestion
    /// The returned string has one line per pseudo header and
    /// one line per TLV (if any).
```

... and remove the next comment line.